ironfish/src/account/accounts.ts

```diff
@@ -630,10 +642,21 @@ export class Accounts {
         }
 
         if (!map.spent) {
+          let confirmedMinimumBlocks = false
+
+          const { blockHash } = transactionMapValue
+          if (blockHash) {
+            const block = await this.chain.getBlock(Buffer.from(blockHash, 'hex'))
+            Assert.isNotNull(block)
+            const confirmations = this.chain.head.sequence - block.header.sequence
+            confirmedMinimumBlocks = confirmations >= minimumBlockConfirmations
```

Comment:
Isn't this code not correct, as we would consider it confirmed even if it's on a fork?

Reply:
Yeah, this could be considered confirmed if on a fork. However, the idea here was choosing a minimum threshold where we have some guarantee of low reversibility after some confirmations. There seems to be similar logic for other projects, where confirmations are checked against a similar threshold when getting [spendable inputs](https://github.com/zcash/zcash/blob/master/src/wallet/wallet.cpp#L2395) or computing [account balance](https://github.com/zcash/zcash/blob/master/src/wallet/wallet.cpp#L4810). These pieces of logic also seem to use the [active chain's height](https://github.com/zcash/zcash/blob/master/src/wallet/wallet.cpp#L6688).

Reply:
This looks different than zcash, where their confirmations are calculated based on 
 depth in the main chain. I also don't think we should consider TX on forks confirmed, because that money is not spendable. Any transactions we create with those notes are going to be considered invalid, even to us.

 - https://github.com/zcash/zcash/blob/master/src/wallet/wallet.cpp#L2393
 - https://github.com/zcash/zcash/blob/master/src/wallet/wallet.cpp#L6672-L6676

Reply:
Added a main chain [check](https://github.com/iron-fish/ironfish/pull/1571/files#diff-975265306b3138d4067a44c1f74a98ef80249217a05ac91dff96236b482c429bR649-R653).